packages/expo-dev-launcher/android/src/expo-45/expo/modules/devlauncher/DevLauncherPackageDelegate.kt

```diff
@@ -20,71 +20,73 @@ import expo.modules.devlauncher.modules.DevLauncherDevMenuExtension
-        override fun onNewIntent(intent: Intent?): Boolean {
-          if (!shouldEnableAutoSetup || intent == null || activityContext == null || activityContext !is ReactActivity) {
-            return false
-          }
-          return DevLauncherController.tryToHandleIntent(activityContext, intent)
-        }
-      }
-    )
+    fun createReactActivityLifecycleListeners(activityContext: Context?): List<ReactActivityLifecycleListener> =
+            listOf(
+                    object : ReactActivityLifecycleListener {
+                        override fun onNewIntent(intent: Intent?): Boolean {
+                            if (!shouldEnableAutoSetup || intent == null || activityContext == null || activityContext !is ReactActivity) {
+                                return false
+                            }
+                            return DevLauncherController.tryToHandleIntent(activityContext, intent)
+                        }
+                    }
+            )
 
-  fun createReactActivityHandlers(activityContext: Context?): List<ReactActivityHandler> =
-    listOf(
-      object : ReactActivityHandler {
-        override fun onDidCreateReactActivityDelegate(activity: ReactActivity, delegate: ReactActivityDelegate): ReactActivityDelegate? {
-          if (!shouldEnableAutoSetup) {
-            return null
-          }
-          return DevLauncherController.wrapReactActivityDelegate(
-            activity,
-            object : DevLauncherReactActivityDelegateSupplier {
-              override fun get(): ReactActivityDelegate {
-                return delegate
-              }
-            }
-          )
-        }
-      }
-    )
+    fun createReactActivityHandlers(activityContext: Context?): List<ReactActivityHandler> =
+            listOf(
+                    object : ReactActivityHandler {
+                        override fun onDidCreateReactActivityDelegate(activity: ReactActivity, delegate: ReactActivityDelegate): ReactActivityDelegate? {
+                            if (!shouldEnableAutoSetup) {
+                                return null
+                            }
+                            return DevLauncherController.wrapReactActivityDelegate(
+                                    activity,
+                                    object : DevLauncherReactActivityDelegateSupplier {
+                                        override fun get(): ReactActivityDelegate {
+                                            return delegate
+                                        }
+                                    }
+                            )
+                        }
+                    }
+            )
 
-  fun createReactNativeHostHandlers(context: Context): List<ReactNativeHostHandler> = listOf(DevLauncherReactNativeHostHandler(context))
+    fun createReactNativeHostHandlers(context: Context): List<ReactNativeHostHandler> =
+            if (shouldEnableAutoSetup) listOf(DevLauncherReactNativeHostHandler(context))
```

Comment:
It seems like this runs before `onCreate` of MainApplication.

Reply:
You folks can suggest a better check?
Ideally when `enableAutoSetup` is false, one would want the behavior to be same as that of a bare RN app if possible.